src/opm/output/eclipse/Summary.cpp

```diff
@@ -1237,6 +1237,35 @@ inline quantity trans_factors ( const fn_args& args ) {
     return { connPos->trans_factor, measure::transmissibility };
 }
 
+inline quantity d_factors ( const fn_args& args ) {
+    const quantity zero = { 0.0, measure::time };
+
+    if (args.schedule_wells.empty())
+        // No wells.  Before simulation starts?
+        return zero;
+
+    auto xwPos = args.wells.find(args.schedule_wells.front()->name());
+    if (xwPos == args.wells.end())
+        // No dynamic results for this well.  Not open?
+        return zero;
+
+    // Like connection rate we need to look up a connection with offset 0.
+    const size_t global_index = args.num - 1;
+    const auto& connections = xwPos->second.connections;
+    auto connPos = std::find_if(connections.begin(), connections.end(),
+        [global_index](const Opm::data::Connection& c)
+    {
+        return c.index == global_index;
+    });
+
+    if (connPos == connections.end())
+        // No dynamic results for this connection.
+        return zero;
+
+    // Connection "d_factor".
+    return { connPos->d_factor, measure::time };
+}
```

Comment:
This should be time/sm3. Do we have that in the unitsytem or do I need to add it? Or can I use measure::time/measure::volume.

Reply:
> This should be `time/sm3`

Probably `time / surface_volume_gas` instead, I think.  Otherwise you'll have inconsistent units when running a model in the `FIELD` or `LAB` unit conventions.

> Do we have that in the `Unitsystem` or do I need to add it?

There is no such named unit in our list of [named units of `measure`](https://github.com/OPM/opm-common/blob/a8d3dff2a4e959f5919122d98fba15b7133346ad/opm/input/eclipse/Units/UnitSystem.hpp#L43).

> Or can I use `measure::time / measure::volume`.

Sadly you cannot.  We don't have arithmetic operators defined for `UnitSystem::measure`, and in any case it should not be `measure::volume` here.